pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -682,7 +720,13 @@ func (a *index) AddressInformation(addresses sets.String) ([]model.AddressInfo,
 }
 
 func (a *index) ServicesForWaypoint(key model.WaypointKey) []model.ServiceInfo {
+	if key.IsGateway {
```

Comment:
What about the case where we don't have a local copy of a global service and the only service instance (i.e VIP) is in a remote network? The East/West gateway probably shouldn't expose it right?

Reply:
What about if a service exists locally but there are no backing workloads?

Reply:
Then we should expose it, but envoy will send a 503 because there won't be any healthy endpoints